commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -243,4 +253,35 @@ public Handler getHandler() {
             });
         }
     }
+
+    public static class TransferParamsBuilder {
+
+        private Map<String, String> mParams = new HashMap<>();
+
+        public TransferParamsBuilder transferMethodType(@NonNull final String transferMethodType) {
+            mParams.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_TYPE, transferMethodType);
+            return this;
+        }
+
+        public TransferParamsBuilder transferMethodProfileType(@NonNull final String transferMethodProfileType) {
+            mParams.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_PROFILE_TYPE,
+                    transferMethodProfileType);
+            return this;
+        }
+
+        public TransferParamsBuilder transferMethodCountry(@NonNull final String transferMethodCountry) {
+            mParams.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_COUNTRY, transferMethodCountry);
+            return this;
+        }
+
+        public TransferParamsBuilder transferMethodCurrency(@NonNull final String transferMethodCurrency) {
+            mParams.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_CURRENCY,
+                    transferMethodCurrency);
+            return this;
+        }
+
+        public Map<String, String> build() {
+            return mParams;
```

Comment:
"product" and "pgtf" need to be set on the UI SDK side. I'm proposing we add it to the params here.  Thoughts @peter-joseph ?
```
mParams.put(InsightEventTag.InsightEventTagEventParams.PRODUCT, "hyperwallet-android-ui-sdk");
mParams.put(InsightEventTag.InsightEventTagEventParams.PAGE_TECHNOLOGY, "Java");
```

Reply:
for this Iteration it's good enough! thanks Danny for raising this up 👍 , since Insight core cannot actually just naively hardcode it, we can pass it to UI SDK

Reply:
I updated. i think this(builder) approach helping to us. we can add "comp" also here.